Docs: Add `no-await-in-loop` doc (#3)
RDambrosio016/RSLint: docs/rules/errors/no-await-in-loop.md

-->
# no-await-in-loop

> Disallow await inside of loop.

Reviewer: ```suggestion
Disallow await inside of loops.
```

You may want to `await` a promise until it is fulfilled or rejected, inside of loops. In such cases, to take

> performance advantages of concurrency with async operations, you should be careful __not__ to `await` in each

Reviewer: ```suggestion
full advantage of concurrency, you should __not__ `await` the promise in every
```

> iteration; otherwise your async operations would be executed in serial.

Reviewer: ```suggestion
iteration, otherwise your async operations will be executed serially.
```
